infra/src/main/kotlin/fi/fta/geoviite/infra/aspects/GeoviiteController.kt

```diff
@@ -0,0 +1,28 @@
+package fi.fta.geoviite.infra.aspects
+
+import fi.fta.geoviite.infra.logging.apiCall
+import org.aspectj.lang.JoinPoint
+import org.aspectj.lang.annotation.Aspect
+import org.aspectj.lang.annotation.Before
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.annotation.RestController
+
+@Target(AnnotationTarget.CLASS)
+@Retention(AnnotationRetention.RUNTIME)
+@RestController
+annotation class GeoviiteController
+
+@Aspect
+@Component
+class GeoviiteControllerAspect {
+    private val logger: Logger = LoggerFactory.getLogger(this::class.java)
```

Comment:
Tämä tarkoittanee että kaikiin näihin logituksiin tulee luokkana tämä GeoviiteControllerAspect, eikä itse controller-luokan nimeä? Pystyisikö täällä poimimaan sen annotoidun luokan nimen jotenkin?

Reply:
Eli siis
```
... -  INFO --- [io-8080-exec-10] f.f.g.i.c.ChangeTimeController : ...
```
vs
```
... -  INFO --- [io-8080-exec-10] f.f.g.i.a.GeoviiteControllerAspect : ...
```

Reply:
Joo, näin käy. Toistaiseksi se varsinainen luokka mikä annotoitiin päätyy sinne lokiviestiin. Sen annotoidun luokan nimen saa kyllä kaivettua jokaisen kutsun yhteydessä, mutta silloin getLoggeria pitää kutsua jokaisella lokituskutsulla. Voin tutkailla tätä hieman lisää kyllä, sillä tuo aiempi oli lokiviestien puolesta mielekkäämpi.

Reply:
👍 Muutettu lokiformaatti aiempaan, tuo hakee nyt loggerit riippuen luokasta concurrenthashmappiin josta niitä voi sitten noutaa joka kutsulla. Toki tuosta tuli implisiittinen epäsuoruus eli lokissa ei nyt näy viittausta aspektiin millään tavalla.